### PR TITLE
optimize docker image size

### PR DESCRIPTION
**Pull request by subDesTagesMitExtraKaese**
split the dockerfile into two stages: one for compiling and one for building

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
+
+RUN python -m venv /opt/venv
+# Make sure we use the virtualenv:
+ENV PATH="/opt/venv/bin:$PATH"
+
+RUN pip install --no-cache-dir matrix-registration-bot
+
+FROM python:3-slim
+
+COPY --from=compile-image /opt/venv /opt/venv
```
Comment (bziemons) [inline Dockerfile:15]: Copying over virtualenvs might be problematic if the locally stored images `python:3` and `python:3-slim` have different Python versions. Due to caching or bad timing this issue might actually occur and would result in a broken image. Any idea how to handle this case?

**Review by bziemons — APPROVED** [review]
Changes look great :) Thanks!

**Comment by moan0s** [comment]
Did anyone test the dockerfiles? In this case I'd merge without checking myself

**Comment by bziemons** [comment]
I did a quick test run of the Dockerfile in the repo's root dir, but it would be good to know from somebody who actually uses this in production.

**Comment by subDesTagesMitExtraKaese** [comment]
I don't know how to use ansible, but i've got it working with docker-compose on a Pi 4

**Comment by moan0s** [comment]
Okay I tested it now and everything looks good. Thanks for the work on this and sorry for the long delay in merging this!

**Comment by moan0s** [comment]
Merged! (via my local development branch)

**Pull request closed without merging**